### PR TITLE
Lint Relay & GraphQL

### PR DESCRIPTION
**Pull request by plasticine**
**tl;dr** Add [`eslint-plugin-graphql`](https://github.com/apollographql/eslint-plugin-graphql) & [`eslint-plugin-relay`](https://github.com/relayjs/eslint-plugin-relay) to our ESLint configuration.

***

Our Relay/GraphQL implementation is super critical to ensuring that the frontend continues to integrate with the Buildkite graph server correctly, be easy to work with and refactor, and generally remain Totes Legit™.

Using `eslint-plugin-graphql` & `eslint-plugin-relay` helps us do this by catching;

- drift from the server schema by catching removed or invalid fields
- the use of deprecated fields
- unused fields meaning we don’t request things we don’t use
- unnamed queries and mutations so that we have better logging on the server
- ...and heeeeaps of other things we can incrementally adopt as we get off classic & clean things up

### Examples;

screen shot 2018-10-25 at 3 10 12 pm
screen shot 2018-10-25 at 3 10 17 pm

screen shot 2018-10-25 at 8 42 25 pm

screen shot 2018-10-25 at 8 43 17 pm

screen shot 2018-10-25 at 8 48 20 pm

screen shot 2018-10-25 at 8 50 55 pm

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+const schemaJson = require('./app/graph/schema.json');
```
Comment (plasticine) [inline .eslintrc.js]: Had to convert this from YAML -> to JS so we can require in our schema JSON.

File: app/components/agent/Show.js [hunk 2]
```diff
@@ -445,6 +445,14 @@ class AgentShow extends React.Component {
   }
 }
 
+/*
+Disabling graphql/no-deprecated-fields here as I think there is some changes probably
+required in the graph implementation so that we can actually use the alternatives to
+`pingedAt`, `stoppedAt`, & `stoppedBy` fields as at the moment it seems like using these
+fields will mean we need do a lot of extra fetching and checking on the client which
+seems bad?
```
Comment (plasticine) [inline app/components/agent/Show.js]: Thoughts welcome. This seems to me like the pragmatic way to get this shipped and catching new errors.

**Comment by toolmantim** [comment]
Rad!
